src/vector/jitsi/index.ts

```diff
@@ -142,25 +143,61 @@ let skipOurWelcomeScreen = false;
                 logger.log("Got OpenID Connect token");
             }
 
-            // TODO: register widgetApi listeners for PTT controls (https://github.com/vector-im/element-web/issues/12795)
-
+            widgetApi.on(`action:${ElementWidgetActions.JoinCall}`,
+                (ev: CustomEvent<IWidgetApiRequest>) => {
+                    ack(ev);
+                    joinConference();
+                },
+            );
             widgetApi.on(`action:${ElementWidgetActions.HangupCall}`,
                 (ev: CustomEvent<IWidgetApiRequest>) => {
-                    if (meetApi) meetApi.executeCommand('hangup');
-                    widgetApi.transport.reply(ev.detail, {}); // ack
+                    ack(ev);
+                    meetApi?.executeCommand('hangup');
+                },
+            );
+            widgetApi.on(`action:${ElementWidgetActions.MuteAudio}`,
+                async (ev: CustomEvent<IWidgetApiRequest>) => {
+                    ack(ev);
+                    if (meetApi && !await meetApi.isAudioMuted()) {
+                        meetApi.executeCommand('toggleAudio');
+                    }
+                },
+            );
+            widgetApi.on(`action:${ElementWidgetActions.UnmuteAudio}`,
+                async (ev: CustomEvent<IWidgetApiRequest>) => {
+                    ack(ev);
+                    if (meetApi && await meetApi.isAudioMuted()) {
+                        meetApi.executeCommand('toggleAudio');
+                    }
+                },
+            );
+            widgetApi.on(`action:${ElementWidgetActions.MuteVideo}`,
+                async (ev: CustomEvent<IWidgetApiRequest>) => {
+                    ack(ev);
+                    if (meetApi && !await meetApi.isVideoMuted()) {
+                        meetApi.executeCommand('toggleVideo');
+                    }
+                },
+            );
+            widgetApi.on(`action:${ElementWidgetActions.UnmuteVideo}`,
+                async (ev: CustomEvent<IWidgetApiRequest>) => {
+                    ack(ev);
+                    if (meetApi && await meetApi.isVideoMuted()) {
+                        meetApi.executeCommand('toggleVideo');
+                    }
                 },
             );
             widgetApi.on(`action:${ElementWidgetActions.StartLiveStream}`,
                 (ev: CustomEvent<IWidgetApiRequest>) => {
                     if (meetApi) {
+                        ack(ev);
```

Comment:
the widget API is expected to handle errors in these sorts of cases - can we move the ack to the bottom of all the handlers?

Reply:
Actually, it looks like moving them below the `await` points in the mute handlers is causing the widget API to spit out (harmless) "Unrecognized action" errors. What's the right thing to do in that case?

Reply:
@turt2live ping in case you missed this